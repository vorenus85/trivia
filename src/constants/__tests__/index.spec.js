import { describe, it, expect, test } from 'vitest'
import { pages, difficulties, categories } from '../index.js'

describe('It should be correct page names', () => {
  test.each`
    name                | value
    ${'start'}          | ${'START'}
    ${'difficulty'}     | ${'DIFFICULTY'}
    ${'questions'}      | ${'QUESTIONS'}
    ${'result'}         | ${'RESULT'}
    ${'detailedResult'} | ${'DETAILED_RESULTS'}
  `('page: $name, should be: $value', ({ name, value }) => {
    const actualResult = pages[name]

    expect(actualResult).toBe(value)
  })

  it('pages matches the array snapshot', () => {
    expect(pages).toMatchSnapshot()
  })
})

describe('difficulties', () => {
  test('should contain three difficulty levels', () => {
    expect(difficulties).toHaveLength(3)
  })

  test('should have correct properties for each difficulty level', () => {
    const expectedDifficulties = [
      { key: 'easy', title: 'Easy', questions: 8, time: 12 },
      { key: 'medium', title: 'Medium', questions: 10, time: 10 },
      { key: 'hard', title: 'Hard', questions: 12, time: 8 }
    ]

    expectedDifficulties.forEach((expected, index) => {
      const difficulty = difficulties[index]
      expect(difficulty.key).toBe(expected.key)
      expect(difficulty.title).toBe(expected.title)
      expect(difficulty.questions).toBe(expected.questions)
      expect(difficulty.time).toBe(expected.time)
    })
  })

  test('should have time and questions as numbers', () => {
    difficulties.forEach((difficulty) => {
      expect(typeof difficulty.questions).toBe('number')
      expect(typeof difficulty.time).toBe('number')
    })
  })

  it('difficulties matches the array snapshot', () => {
    expect(difficulties).toMatchSnapshot()
  })
})

describe('Categories tests', () => {
  it('should be 6 quiz categories', () => {
    expect(categories.length).toBe(6)
  })

  it('quiz categories matches the array snapshot', () => {
    const quizCategories = categories

    expect(quizCategories).toMatchSnapshot()
  })
})
