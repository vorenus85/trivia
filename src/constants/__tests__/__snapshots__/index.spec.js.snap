// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Categories tests > quiz categories matches the array snapshot 1`] = `
[
  {
    "key": "9",
    "title": "General Knowledge",
  },
  {
    "key": "27",
    "title": "Animals",
  },
  {
    "key": "23",
    "title": "History",
  },
  {
    "key": "17",
    "title": "Science & Nature",
  },
  {
    "key": "21",
    "title": "Sports",
  },
  {
    "key": "14",
    "title": "TV Series",
  },
]
`;

exports[`It should be correct page names > pages matches the array snapshot 1`] = `
{
  "detailedResult": "DETAILED_RESULTS",
  "difficulty": "DIFFICULTY",
  "questions": "QUESTIONS",
  "result": "RESULT",
  "start": "START",
}
`;

exports[`difficulties > difficulties matches the array snapshot 1`] = `
[
  {
    "key": "easy",
    "questions": 8,
    "time": 12,
    "title": "Easy",
  },
  {
    "key": "medium",
    "questions": 10,
    "time": 10,
    "title": "Medium",
  },
  {
    "key": "hard",
    "questions": 12,
    "time": 8,
    "title": "Hard",
  },
]
`;
